doc-examples/src/main/java/jms/javasamples/JmsToHttpGet.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package jms.javasamples;
+
+// #sample
+import akka.Done;
+import akka.actor.ActorSystem;
+import akka.http.javadsl.Http;
+import akka.http.javadsl.model.HttpRequest;
+import akka.japi.Pair;
+import akka.stream.ActorMaterializer;
+import akka.stream.KillSwitch;
+import akka.stream.Materializer;
+import akka.stream.alpakka.jms.JmsConsumerSettings;
+import akka.stream.alpakka.jms.JmsProducerSettings;
+import akka.stream.alpakka.jms.javadsl.JmsConsumer;
+import akka.stream.alpakka.jms.javadsl.JmsProducer;
+import akka.stream.javadsl.Keep;
+import akka.stream.javadsl.Sink;
+import akka.stream.javadsl.Source;
+import akka.util.ByteString;
+
+// #sample
+
+import playground.ActiveMqBroker;
+import playground.WebServer;
+import scala.concurrent.ExecutionContext;
+
+import javax.jms.ConnectionFactory;
+import java.util.Arrays;
+import java.util.concurrent.CompletionStage;
+
+public class JmsToHttpGet {
+
+  public static void main(String[] args) throws Exception {
+    JmsToHttpGet me = new JmsToHttpGet();
+    me.run();
+  }
+
+  private final ActorSystem system = ActorSystem.create();
+  private final Materializer materializer = ActorMaterializer.create(system);
+  private final ExecutionContext ec = system.dispatcher();
+
+  private void enqueue(ConnectionFactory connectionFactory, String... msgs) {
+    Sink<String, ?> jmsSink =
+        JmsProducer.textSink(
+            JmsProducerSettings.create(connectionFactory).withQueue("test")
+        );
+    Source.from(Arrays.asList(msgs)).runWith(jmsSink, materializer);
+  }
+
+
+  private void run() throws Exception {
+    ActiveMqBroker activeMqBroker = new ActiveMqBroker();
+    activeMqBroker.start();
+
+    WebServer webserver = new WebServer();
+    webserver.start("localhost", 8080);
+
+
+    ConnectionFactory connectionFactory = activeMqBroker.createConnectionFactory();
+    enqueue(connectionFactory, "a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k");
+    // #sample
+
+    final Http http = Http.get(system);
+
+    Source<String, KillSwitch> jmsSource =        // (1)
+        JmsConsumer.textSource(
+            JmsConsumerSettings.create(connectionFactory)
+                .withBufferSize(10)
+                .withQueue("test")
+        );
+
+
+    Pair<KillSwitch, CompletionStage<Done>> pair =
+        jmsSource                                                  //: String
+            .map(ByteString::fromString)                           //: ByteString   (2)
+            .map(bs ->
+                HttpRequest.create("http://localhost:8080/hello")
+                    .withEntity(bs)
+            )                                                      //: HttpRequest  (3)
+            .mapAsyncUnordered(4, http::singleRequest)             //: HttpResponse (4)
```

Comment:
Here and below move the argument to the variable so it is more clear what this number does.